Conductor/Client/Interfaces/IWorkflowTask.cs

```diff
@@ -1,12 +1,15 @@
 ﻿using Conductor.Client.Models;
 using Conductor.Client.Worker;
+using System.Threading;
+using System.Threading.Tasks;
 
 namespace Conductor.Client.Interfaces
 {
     public interface IWorkflowTask
     {
         string TaskType { get; }
         WorkflowTaskExecutorConfiguration WorkerSettings { get; }
-        TaskResult Execute(Task task);
+        Task<TaskResult> Execute(Models.Task task, CancellationToken token = default);
+        TaskResult Execute(Models.Task task);
```

Comment:
Can you mark this as deprecated so its clear that the users should not use this going forward?
@BommannanRa

Reply:
Yes @v1r3n We have added deprecate message to the code. When someone try using Execute method, it would show message to move to ExecuteAsync (new implementation).

Reply:
Hi @BommannanRa let's not change the method name as we want to keep the name consistent with the previous implementation `ExecuteAsync` should be just `Execute`